clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -244,11 +246,11 @@ public void updateFetchPositions(Set<TopicPartition> partitions) {
                 continue;
 
             if (subscriptions.isOffsetResetNeeded(tp)) {
-                resetOffset(tp);
+                resetOffsets(Collections.singleton(tp));
             } else if (subscriptions.committed(tp) == null) {
                 // there's no committed position, so we need to reset with the default strategy
                 subscriptions.needOffsetReset(tp);
-                resetOffset(tp);
+                resetOffsets(Collections.singleton(tp));
```

Comment:
I guess there's still room to batch the offset lookup here. If it's too much trouble here, maybe we can open a new JIRA so that we don't forget about it?